pandas/tests/arithmetic/test_datetime64.py

```diff
@@ -1810,55 +1810,50 @@ def test_dt64ser_sub_datetime_dtype(self):
     # TODO: This next block of tests came from tests.series.test_operators,
     # needs to be de-duplicated and parametrized over `box` classes
 
-    def test_operators_datetimelike_invalid(self, all_arithmetic_operators):
-        # these are all TypeEror ops
+    @pytest.mark.parametrize(
+        "left, right, op_fail",
+        [
+            [
+                [Timestamp("20111230"), Timestamp("20120101"), NaT],
+                [Timestamp("20111231"), Timestamp("20120102"), Timestamp("20120104")],
+                ["__sub__", "__rsub__"],
+            ],
+            [
+                [Timestamp("20111230"), Timestamp("20120101"), NaT],
+                [timedelta(minutes=5, seconds=3), timedelta(minutes=5, seconds=3), NaT],
+                ["__add__", "__radd__", "__sub__"],
+            ],  # TODO(jreback) __rsub__ should raise?
+            [
+                [
+                    Timestamp("20111230", tz="US/Eastern"),
+                    Timestamp("20111230", tz="US/Eastern"),
+                    NaT,
+                ],
+                [timedelta(minutes=5, seconds=3), NaT, timedelta(minutes=5, seconds=3)],
+                ["__add__", "__radd__", "__sub__"],
+            ],
+        ],
+    )
+    def test_operators_datetimelike_invalid(
+        self, request, left, right, op_fail, all_arithmetic_operators
+    ):
+        # these are all TypeError ops
         op_str = all_arithmetic_operators
-
-        def check(get_ser, test_ser):
-
-            # check that we are getting a TypeError
-            # with 'operate' (from core/ops.py) for the ops that are not
-            # defined
-            op = getattr(get_ser, op_str, None)
-            # Previously, _validate_for_numeric_binop in core/indexes/base.py
-            # did this for us.
-            with pytest.raises(
-                TypeError, match="operate|[cC]annot|unsupported operand"
-            ):
-                op(test_ser)
-
-        # ## timedelta64 ###
-        td1 = Series([timedelta(minutes=5, seconds=3)] * 3)
-        td1.iloc[2] = np.nan
-
-        # ## datetime64 ###
-        dt1 = Series(
-            [Timestamp("20111230"), Timestamp("20120101"), Timestamp("20120103")]
-        )
-        dt1.iloc[2] = np.nan
-        dt2 = Series(
-            [Timestamp("20111231"), Timestamp("20120102"), Timestamp("20120104")]
+        request.node.add_marker(
+            pytest.mark.xfail(
+                op_str in op_fail, reason=f"{op_str} doesn't raise TypeError"
```

Comment:
can you separate out the `op_str in op_fail` check?  took me a sec to realize this was a conditional xfail.

Reply:
or instead of xfailing, could do:

```
if op not in op_fail:
    with pytest.raises...
else:
    op(arg2)  # smoketest that it doesnt raise
```